op-node/p2p/discovery.go

```diff
@@ -11,6 +11,8 @@ import (
 	"net"
 	"time"
 
+	"github.com/btcsuite/btcd/btcec/v2"
+	decredSecp "github.com/decred/dcrd/dcrec/secp256k1/v4"
```

Comment:
Why decred, btcd and geth ec?

Reply:
Summary of the secp256k1 Go crypto situation:
- libp2p updated from btcd/btcec to btcd/btcec v2
- btcd/btcec v1 wrapped Go ecdsa
- btcd/btcec v2 wraps decred
- decred is faster, and thus btcd updated
- btcd earlier also went from monolithic releases to per-module go releases, and had versioning mistakes in v1. These were fixed with beta versions / new releases eventually, but end-users started looking to update to v2.
- geth uses a C library for secp256k1, and uses Go ecdsa curve types to integrate that into Go (but use their own instance of the curve definitions, same values, but not same memory address, making shallow equality checks not work...)
- decred has functions to convert to Go ecdsa types
- After type casting the general libp2p pubkey/privkey interface to a concrete type, it can be casted to the underlying types (btcd, decred) to use their functionality to convert to Go ecdsa, and then you can swap the curve instance to point to the one by geth for the shallow equality checks in the Geth discovery v5 package to work.